admin_manual/installation/caddy.rst

```diff
@@ -0,0 +1,67 @@
+===================
+Caddy configuration
+===================
+
+.. warning::
+    Please note that webservers other than Apache 2.x are not officially supported.
+
+.. note::
+    This page covers example Caddy configuration to run a Nextcloud server.
+    These configurations examples were originally provided by
+    `@ntninja <https://ninetailed.ninja>`_ based on the :doc:`nginx` sample and
+    are exclusively community-maintained. (Thank you contributors!)
+
+-  This guide assumes you are using Caddy 2.6 or later and the presented sample
+   configuration will not work on older versions without modification.
+-  Caddy takes care of TLS certificate configuration and HTTP-to-HTTPS redirects
+   automatically, so that is not covered here.
+-  The example configuration makes use of the `route <https://caddyserver.com/docs/caddyfile/directives/route>`_
+   directive which disables all directive reordering usually done by Caddy. This
+   means that anything within that block should be read strictly top-to-bottom
+   unlike what you may be used from NGINX or regular (non-route) Caddy
+   configurations.
+-  Be careful about line breaks if you copy the examples, as long lines may be
+   broken for page formatting.
+-  Some environments might need a ``cgi.fix_pathinfo`` set to ``1`` in their
+   ``php.ini``.
+
+Nextcloud in the webroot of Caddy
+---------------------------------
+
+The following configuration should be used when Nextcloud is placed in the
+webroot of your nginx installation. In this example it is
+``/var/www/nextcloud`` and it is accessed via ``http(s)://cloud.example.com/``
+
+.. literalinclude:: Caddyfile.sample
+   :language: caddy
+
+..
+   Nextcloud in a subdir of the Caddy webroot
+   ------------------------------------------
+
+   This section remains to be written, but should be as simple as wrapping most
```

Comment:
I have done tests with this. The problem is that handle_path /next/cloud/* does only remove the prefix from the CGI variable PATH_INFO, but not from REQUEST_URI. While this is apparently what the spec requires, it breaks Nextcloud.

My current understanding for adding a subdirectory comprises:

    set in the app (fpm) container OVERWRITEWEBROOT=/next/cloud
    wrap most of the Caddyfile in a handle_file /next/cloud/* environment or use uri strip_prefix /next/cloud
    set before php_fastcgi directive a variable: vars rewritten_uri {uri}
    add in the environment of php_fastcgi this env rewrite: env REQUEST_URI {vars.rewritten_uri}

3-4 should then look like this:

vars rewritten_uri {uri}

~~~
# Let everything else be handled by the PHP-FPM component
php_fastcgi app:9000 {
  env modHeadersAvailable true # Avoid sending the security headers twice
  env front_controller_active true # Enable pretty urls
  env REQUEST_URI {vars.rewritten_uri} # Nextcloud relies on REQUEST_URI instead of rewritten PATH_INFO
}
~~~

An alternative approach is to use Caddy two times. One for SSL off-loading and striping of the path, one to serve static content and call php-fpm.


References:
- https://caddy.community/t/how-to-change-request-uri-passed-to-fastcgi-script/11533
- https://caddy.community/t/rewrite-first-then-evaluate-php-fastcgi/15164/11
- https://github.com/caddyserver/caddy/issues/5312#issuecomment-1445254932

Reply:
(Just so you know, I’m not planning on testing this / extending the docs in this regard myself. You’ll need to make your own PR if you’d like to do that.)